src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,97 @@
+package duke;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.command.ClearCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnmarkCommand;
+import duke.enums.Commands;
+import duke.enums.Views;
+
+/**
+ * Parses the input given from the user and returns the appropriate commands
+ */
+public class Parser {
+    /**
+     * Given the user input, sort it out and returns the command associated with the
+     * input
+     *
+     * @param input string from the user
+     * @return Command object that matches the user input
+     * @throws DukeException
+     */
+    public static Command parse(String input) throws DukeException {
+        switch (input) {
+        case "list":
+            return new ListCommand();
+        case "clear":
+            return new ClearCommand();
+        case "bye":
+        case "exit":
+            return new ExitCommand();
+        default:
+            // Have to do it at the starts with because what if "todo mark this as done"
+            if (input.startsWith(Commands.MARK.cmd())) {
+                int taskNo = getNumbers(input) - 1;
+                return new MarkCommand(taskNo);
+            } else if (input.startsWith(Commands.UNMARK.cmd())) {
+                int taskNo = getNumbers(input) - 1;
+                return new UnmarkCommand(taskNo);
+            } else if (input.startsWith(Commands.TODO.cmd())) {
+                String title = input.substring(Commands.TODO.cmd().length());
+                return new TodoCommand(title);
+            } else if (input.startsWith(Commands.DEADLINE.cmd())) {
+                if (input.indexOf(Commands.BY.cmd()) == -1) {
+                    throw new DukeException(Views.MISSING_ARGS_ERR_STRING.eng());
+                }
+                String title = input.substring(Commands.DEADLINE.cmd().length(), input.indexOf(Commands.BY.cmd()));
+                String deadline = input.substring(input.indexOf(Commands.BY.cmd()));
+                return new DeadlineCommand(title, deadline);
+            } else if (input.startsWith(Commands.EVENT.cmd())) {
+                if (input.indexOf(Commands.FROM.cmd()) == -1 || input.indexOf(Commands.TO.cmd()) == -1) {
+                    throw new DukeException(Views.MISSING_ARGS_ERR_STRING.eng());
+                }
+                String title = input.substring(Commands.EVENT.cmd().length(), input.indexOf(Commands.FROM.cmd()));
+                String from = input.substring(input.indexOf(Commands.FROM.cmd()), input.indexOf(Commands.TO.cmd()));
```

Comment:
Some of the lines (line 56, 63, 64) have a line length more than the soft limit of 110 characters. Although it is still less than the hard limit of 120 characters, perhaps the lines could be separated with a line break to improve readability?